src/main/java/cloud/eppo/api/IAssignmentCache.java

```diff
@@ -0,0 +1,15 @@
+package cloud.eppo.api;
+
+import cloud.eppo.logging.Assignment;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public interface IAssignmentCache {
+  void put(@NotNull String key, @NotNull Assignment assignment);
+
+  @Nullable Assignment get(@NotNull String key);
```

Comment:
I wonder if we should explicitly have it take a `flagKey`, `subjectKey`, `allocationKey`, and `variationKey` vs. a single string.